### ISSUE TITLE
The converter doesn't take the color attributes

### ISSUE THREAD

**Issue by CatHood0** [VisualSystemsCorp/vsc_quill_delta_to_html#8]
When I use deltas like this: `[{"insert":"text"},{"insert":" "},{"insert":"with a color","attributes":{"color":"0x0FF ","header":"2"}}` The conversion doesn't take the color attribute, but the header does

**Comment by dsyrstad** [comment]
The color value has to be valid. Quill supports two hex notations starting with a `#` followed by three or six RGB hex digits - e.g. `#0ff` or `#00ffff` (see https://quilljs.com/docs/delta/#document and https://github.com/VisualSystemsCorp/vsc_quill_delta_to_html/blob/5a5b7035dcc168c2df51407aa5f26076ea5df4a9/lib/src/op_attribute_sanitizer.dart#L368). It doesn't support `0x...` for a hex color. 

It also supports an `rgb(redvalue, greenvalue, bluevalue)` value (https://github.com/VisualSystemsCorp/vsc_quill_delta_to_html/blob/5a5b7035dcc168c2df51407aa5f26076ea5df4a9/lib/src/op_attribute_sanitizer.dart#L377).

It also supports a CSS color literal name like `black` or `red` (https://github.com/VisualSystemsCorp/vsc_quill_delta_to_html/blob/5a5b7035dcc168c2df51407aa5f26076ea5df4a9/lib/src/op_attribute_sanitizer.dart#L373).

**Comment by JouseLovic** [comment]
Thank you very much for the quick response. I had previously seen those functions, but I thought that not using the attribute might be causing a failure somewhere. I would like to be able to add that functionality (especially because it would be complicated if I tried to manually modify each of my deltas and transform the colors into valid hex values). What points should I keep in mind when trying to add this new function to validate hex values that start with `0x...`"

**Comment by JouseLovic** [comment]
I had seen those functions before, but I thought that not using the attribute might be causing a failure somewhere. Now I know where the error is (I've been scratching my head for a while because it wasn't taking the attribute). Thank you very much for the help!

**Comment by dsyrstad** [comment]
@JouseLovic If you want to handle different values for `color`, you can do so by customizing the inline style rendering. See https://pub.dev/packages/vsc_quill_delta_to_html#rendering-inline-styles

**Comment by Mamena2020** [comment]
I got same issue. i'm using [Flutter Quill](https://pub.dev/packages/flutter_quill#conversion-to-html) and convert to delta. And then when convert to html, the color attributes not include. this is the color generate from flutter quill `#FF00B0FF` , `#FFFF5797`.
 
- delta
```
    {insert: testing, attributes: {color: #FF00B0FF}}
``` 
- convert code

``` dart
    final deltaJson = controller.document
                                          .toDelta()
                                          .toJson();
    final converter = QuillDeltaToHtmlConverter(
                                        List.castFrom(deltaJson),
                                        ConverterOptions.forEmail()
                                          ..converterOptions.inlineStylesFlag =
                                              true
                                          ..converterOptions
                                              .allowBackgroundClasses = true
                                          ..converterOptions.inlineStyles =
                                              InlineStyles({
                                            'color': InlineStyleType(
                                              fn: (value, op) {
                                                return 'color:$value';
                                              },
                                            ),
                                          }),
                                      );
```
but still cannot convert color attribute, any solution for this?

**Comment by CatHood0** [comment]
@Mamena2020 Flutter quill doesn't let changes the color options of toolbar buttons?

**Comment by CatHood0** [comment]
I only fix that issue because i use AppFlowyEditor that let me change color options (I must to create a custom button for implement it).

And i found a "error" (is only a error by me, because i need different transformation of that attributes), when there color attributes between spaces, and, if after the color attributes block, comes a space or word, that not has any attribute, the converter transform that in something like:

`<p><p style="color:rgb(239,39,4)">colored word</p> word<p style="color:rgb(239,39,4)">another colored word</p></p>`

I prefer something as: `<p style="color:rgb(239,39,4)">colored word</p><p>space</p><p style="color:rgb(239,39,4)">another colored word</p>` or `<span style="color:rgb(239,39,4)">colored word</span><p>space</p><span style="color:rgb(239,39,4)">another colored word</span>

**Comment by Mamena2020** [comment]
@CatHood0 so your saying that, output color from flutter quill like `#FF00B0FF` not working? and have to make custom color picker with format RGB?

**Comment by CatHood0** [comment]
Apparently no. I guess you should change output color from HEX to RGB.However, i make a little change in `op_to_html_converter` for detect (i think that's work) not only literal colors, after that changes rgb works for my

Exactly this: 
```dart
    var props = propsArr
        .where((prop) =>
            isTruthy(attrs[prop]) &&
            (prop != 'background' ||
                OpAttributeSanitizer.isValidColorLiteral(attrs[prop])))
        .map((prop) => '$prop-${attrs[prop]}')
        .toList();
```

To this: 
```dart
 var props = propsArr
        .where((prop) =>
            isTruthy(attrs[prop]) &&
            (prop != 'background' ||
                OpAttributeSanitizer.isValidColorLiteral(attrs[prop]) ||
                OpAttributeSanitizer.isValidHexColor(attrs[prop]) ||
                OpAttributeSanitizer.isValidRGBColor(attrs[prop])))
        .map((prop) => '$prop-${attrs[prop]}')
        .toList();
```

**Comment by Mamena2020** [comment]
@CatHood0 finally it works after changing color from hex to rgb.

Flutter quill delta to html.
``` dart
    // controller  => QuillController 
    final delta = controller.document.toDelta();
    final html = deltaToHtml(delta: delta);
```


``` dart
    String? deltaToHtml({required Delta delta}) {
        try {
          final deltaJson = deltaHexColorToRGB(deltaJson: delta.toJson());

          final converter = QuillDeltaToHtmlConverter(
              List.castFrom(deltaJson),
              ConverterOptions.forEmail()
                ..converterOptions.inlineStylesFlag = true
                ..converterOptions.allowBackgroundClasses = true
              );
          return converter.convert();
        } catch (error) {
          if (kDebugMode) {
            print(error);
          }
        }
        return null;
      }
```
change hex color to rgb
``` dart
    List<dynamic> deltaHexColorToRGB({required List<dynamic> deltaJson}) {
        for (var d in deltaJson) {
          if (d['attributes'] != null && d['attributes']['color'] != null) {
            String hexColor =
                d['attributes']['color'].toString().replaceAll('#', '');
            var color = Color(int.parse('0x$hexColor'));
            d['attributes']['color'] =
                'rgb(${color.red},${color.green},${color.blue})';
          }
        }
        return deltaJson;
      }
```

**Comment by CatHood0** [comment]
I'm glad you made it, that solution is good too

**Comment by dsyrstad** [comment]
Addressed with PR #16
